pages/docs/couple-your-code/couple-your-code-porting-v2-3.md

```diff
@@ -113,39 +113,46 @@ Please add breaking changes here when merged to the `develop` branch.
-  - `PRECICE_PythonActions` -> `PRECICE_FEATURE_PYTHON_ACTIONS`
-  - `PRECICE_ENABLE_C` -> `PRECICE_BINDINGS_C`
-  - `PRECICE_ENABLE_FORTRAN` ->`PRECICE_BINDINGS_FORTRAN`
-  - `PRECICE_ENABLE_LIBBACKTRACE`  -> `PRECICE_FEATURE_LIBBACKTRACE_STACKTRACES`
+- Implicit coupling
+  - Remove `precice::constants::actionReadIterationCheckpoint()` and `precice::constants::actionWriteIterationCheckpoint()`
+  - Remove `markActionFulfilled()` of the above actions.
+  - Replace `isActionRequired()` of the above actions with `requiresReadingCheckpoint()` or `requiresWritingCheckpoint()`.
+- Migrate data access
+    - Replace the commands to read data: `readBlockVectorData`, `readVectorData`, `readBlockScalarData`, `readScalarData` with a single command `readData`.
+    - Replace the commands to write data: `writeBlockVectorData`, `writeVectorData`, `writeBlockScalarData`, `writeScalarData` with a single command `writeData`.
+    - Replace the commands to write gradient data: `writeBlockVectorGradientData`, `writeVectorGradientData`, `writeBlockScalarGradientData`, `writeScalarGradientData` with a single command `writeGradientData`.
+    - The signature of `readData`, `writeData` and `writeGradientData` has changed from `const int*`, `const double*`, and `double*` to `span<const VertexID>`, `span<const double>`, and `span<double>`. The sizes of passed spans are checked by preCICE. spans can be constructed using a pointer and size, or by a container providing `.data()` and `.size()`. Examples for the latter are `std::vector`, `std:array`, and `Eigen::VectorXd`.
+  - To simplify migration, use `getMaxTimeStepSize()` as relative read time for now and read up on time interpolation later.
+- Migrate data initialization
+  - Remove `precice::constants::actionWriteInitialData()`.
+  - Remove `markActionFulfilled()` of write initial data.
+  - Replace `isActionRequired()` of write initial data with `requiresInitialData()`.
+  - Remove `initializeData()`. The function `initializeData()` has been merged into `ìnitialize()`.
+  - Move the data initalization before the call to `initialize()`. You have to initialize the data if `requiresInitialData()` returns `true`.
+- Rename the functions:
+  - Replace `getMeshVerticesAndIDs` with `getMeshVertexIDsAndCoordinates`. Change the input argument meshID to meshName and swap the arguments.
+  - Replace `isMeshConnectivityRequired` with `requiresMeshConnectivityFor`. Instead of the input argument `meshID`, pass the `meshName`.
+  - Replace `isGradientDataRequired` with `requiresGradientDataFor`. Instead of the input argument `dataID`, pass the `meshName` and `dataName`.
+- Remove the following without a replacement:
+  - Remove `mapWriteDataFrom()` and `mapReadDataTo()`.
+  - Remove `hasMesh()` and `hasData()`.
+  - Remove `hasToEvaluateSurrogateModel()` and `hasToEvaluateFineModel()`.
+  - Remove `getMeshVertices()` and `getMeshVertexIDsFromPositions()`. This information is already known by the adapter.
```

Comment:
Since this was not trivial in some cases, you can link to the respective examples page.

Reply:
```suggestion
  - Remove `getMeshVertices()` and `getMeshVertexIDsFromPositions()`. This information is already known by the adapter. We docummented [strategies on how to handle this](couple-your-code-defining-mesh-connectivity.html) in adpters.
```